perses/rjmc/geometry.py

```diff
@@ -2088,47 +2114,188 @@ def _determine_extra_torsions(self, torsion_force, reference_topology, growth_in
             The torsion force with extra torsions added appropriately.
 
         """
+        from perses.rjmc import coordinate_numba
         # Do nothing if there are no atoms to grow.
         if len(growth_indices) == 0:
             return torsion_force
 
+        #set ring restraints
+        _logger.debug(f"\t\t\tattempting to add ring restraints")
+
         #get the list of torsions in the molecule that are not about a rotatable bond
         # Note that only torsions involving heavy atoms are enumerated here.
         rotor = oechem.OEIsRotor()
         torsion_predicate = oechem.OENotBond(rotor)
-        non_rotor_torsions = list(oechem.OEGetTorsions(oemol, torsion_predicate))
+        non_rotor_torsions = list(oechem.OEGetTorsions(self.networkx_structure.mol_oemol, torsion_predicate))
```

Comment:
note that we use the `networkx_strucutre` graph to query oechem.Atom objects for chirality